src/mode.js

```diff
@@ -58,24 +57,19 @@ export function getMode(opt_win) {
  * @return {!ModeDef}
  */
 function getMode_(win) {
-  const hashQuery = parseQueryString(
-    // location.originalHash is set by the viewer when it removes the fragment
-    // from the URL.
-    win.location['originalHash'] || win.location.hash
-  );
+  const hashParams = getHashParams(win);
 
   // The `minified`, `test` and `localDev` properties are replaced
   // as boolean literals when we run `amp dist` without the `--fortesting`
   // flags. This improved DCE on the production file we deploy as the code
   // paths for localhost/testing/development are eliminated.
   return {
     localDev: coreMode.isLocalDev(win),
-    development: isModeDevelopment(win),
+    development: isModeDevelopment(win, hashParams),
     esm: IS_ESM,
     // amp-geo override
-    geoOverride: hashQuery['amp-geo'],
+    geoOverride: hashParams['amp-geo'],
```

Comment:
We need to 🔪  `geoOverride` next! it does not belong here in mode

Reply:
Already workin' on it in another branch 🙃